src/patternfly/components/CodeSnippet/code-snippet.scss

```diff
@@ -0,0 +1,69 @@
+.pf-c-code-snippet {
+  // action group
+  --pf-c-code-snippet__controls--c-button--m-control--Color: var(--pf-global--Color--200);
+  --pf-c-code-snippet__controls--c-button--m-control--hover--Color: var(--pf-global--Color--100);
+  --pf-c-code-snippet__controls--c-button--m-control--focus--Color: var(--pf-global--Color--100);
+
+  // code
+  --pf-c-code-snippet__code--BorderColor: var(--pf-global--BorderColor--100);
+  --pf-c-code-snippet__code--BorderWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-code-snippet__code--BackgroundColor: var(--pf-global--BackgroundColor--100);
+
+  // tab
+  --pf-c-code-snippet__tab--BackgroundColor: var(--pf-global--BackgroundColor--100);
+  --pf-c-code-snippet__tab--Color: var(--pf-global--Color--200);
+  --pf-c-code-snippet__tab--Top: var(--pf-c-code-snippet__code--before--BorderWidth);
+  --pf-c-code-snippet__tab--PaddingTop: var(--pf-global--spacer--form-element);
+  --pf-c-code-snippet__tab--PaddingRight: var(--pf-global--spacer--sm);
+  --pf-c-code-snippet__tab--PaddingBottom: var(--pf-global--spacer--form-element);
+  --pf-c-code-snippet__tab--PaddingLeft: var(--pf-global--spacer--sm);
+  --pf-c-code-snippet__tab--BorderWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-code-snippet__tab--BorderTopColor: var(--pf-global--BorderColor--100);
+  --pf-c-code-snippet__tab--BorderRightColor: var(--pf-global--BorderColor--100);
+  --pf-c-code-snippet__tab--BorderBottomColor: var(--pf-global--BackgroundColor--100);
+  --pf-c-code-snippet__tab--BorderLeftColor: var(--pf-global--BorderColor--100);
+
+  // control
+  --pf-c-code-snippet__header--Top: var(--pf-c-code-snippet__tab--BorderWidth);
+
+  width: 100%;
+}
+
+.pf-c-code-snippet__header {
+  position: relative;
+  top: var(--pf-c-code-snippet__header--Top);
```

Comment:
This creates a `1px` gap above the header - can you use a negative margin instead?

Though I wonder if we want to do that, or if we should just leave this element's positioning normal, then use a negative margin on the `__tab` and potentially the `__controls` element since @mceledonia mentioned maybe moving the controls. It seems like only the `__tab` is the element that should overlay on top of the editor.

Reply:
Yeah if design thinks we should go with what michael suggested, then i could see us just adding it to the tabs. but for now i think they will both need it. i think it makes sense to add it separately to both tab and controls so we can be flexible with how we change it in the future ?

<img width="561" alt="Screen Shot 2020-09-09 at 4 35 27 PM" src="https://user-images.githubusercontent.com/20118816/92650459-88babe00-f2ba-11ea-8e9a-4d199d291408.png">
<img width="471" alt="Screen Shot 2020-09-09 at 4 35 32 PM" src="https://user-images.githubusercontent.com/20118816/92650460-89535480-f2ba-11ea-9cb5-1053c0127183.png">